### PR TITLE
Add framework detection & logging

### PR DESCRIPTION
**Pull request by william-msft**
- [x] The purpose of this PR is explained in this message or in an issue. If an issue please include a reference as https://devdiv.visualstudio.com/DevDiv/_boards/board/t/Oryx/Stories/?workitem=1357977
- [ ] Tests are included and/or updated for code changes.
- [ ] Proper license headers are included in each file.

### CODE REVIEW

File: src/BuildScriptGenerator/DotNetCore/DotnetCorePlatform.cs [hunk 1]
```diff
@@ -138,6 +138,8 @@ public PlatformDetectorResult Detect(RepositoryContext context)
             if (dotNetCorePlatformDetectorResult.InstallAOTWorkloads)
             {
                 installBlazorWebAssemblyAOTWorkloadCommand = DotNetCoreConstants.InstallBlazorWebAssemblyAOTWorkloadCommand;
+                manifestFileProperties["Frameworks"] = "blazor";
```
Comment (gpcastro) [inline src/BuildScriptGenerator/DotNetCore/DotnetCorePlatform.cs:141]: use consts for those strings

File: src/BuildScriptGenerator/Hugo/HugoPlatform.cs [hunk 2]
```diff
@@ -79,6 +79,8 @@ public PlatformDetectorResult Detect(RepositoryContext context)
         {
             var manifestFileProperties = new Dictionary<string, string>();
             manifestFileProperties[ManifestFilePropertyKeys.HugoVersion] = detectorResult.PlatformVersion;
+            manifestFileProperties["Frameworks"] = "hugo";
```
Comment (gpcastro) [inline src/BuildScriptGenerator/Hugo/HugoPlatform.cs:82]: same: use consts

File: src/BuildScriptGenerator/Ruby/RubyPlatform.cs [hunk 3]
```diff
@@ -113,6 +113,11 @@ public PlatformDetectorResult Detect(RepositoryContext context)
             }
 
             var buildProperties = new Dictionary<string, string>();
+            if (RubyConstants.ConfigYmlFileName != null)
+            {
+                buildProperties["Frameworks"] = "jekyll";
```
Comment (qianz2) [inline src/BuildScriptGenerator/Ruby/RubyPlatform.cs:118]: we can create a constant class to store all framework names

**Review by qianz2 — DISMISSED** [review]

File: src/BuildScriptGenerator/DotNetCore/DotnetCorePlatform.cs (hunk 1 above)
Comment (william-msft) [reply src/BuildScriptGenerator/DotNetCore/DotnetCorePlatform.cs]: Updated

File: src/BuildScriptGenerator/Hugo/HugoPlatform.cs (hunk 2 above)
Comment (william-msft) [reply src/BuildScriptGenerator/Hugo/HugoPlatform.cs]: Updated

**Review by qianz2 — APPROVED** [review]

**Pull request merged**